frankenstein/consul_client.go

```diff
@@ -0,0 +1,168 @@
+package frankenstein
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	consul "github.com/hashicorp/consul/api"
+)
+
+const (
+	longPollDuration = 10 * time.Second
+)
+
+// ConsulClient is a high-level client for Consul, that exposes operations
+// such as CAS and Watch which take callbacks.  It also deals with serialisation.
+type ConsulClient interface {
+	Get(key string, out interface{}) error
+	CAS(key string, out interface{}, f CASCallback) error
+	WatchPrefix(prefix string, out interface{}, done chan struct{}, f func(string, interface{}) bool)
+}
+
+// CASCallback is the type of the callback to CAS.  If err is nil, out must be non-nil.
+type CASCallback func(in interface{}) (out interface{}, retry bool, err error)
+
+// NewConsulClient returns a new ConsulClient
+func NewConsulClient(addr string) (ConsulClient, error) {
```

Comment:
It was originally like this as the interface (which this returns) was above the constructor, and thats all your should care about, right?  But I don't mind.